target/linux/ramips/image/common-sercomm.mk

```diff
@@ -64,6 +91,29 @@ define Build/sercomm-prepend-tagged-kernel
 	mv $(IMAGE_KERNEL).tagged $@
 endef
 
+define Build/sercomm-tag-factory-type-B
+	$(call Build/sercomm-pid,$@.hdrfactory)
+	printf $$(stat -c%s $(IMAGE_KERNEL)) | dd seek=$$((0x70)) of=$@.hdrfactory \
+		bs=1 conv=notrunc 2>/dev/null
+	printf $$(stat -c%s $@) | dd seek=$$((0x80)) of=$@.hdrfactory bs=1 \
+		conv=notrunc 2>/dev/null
+	cat $(IMAGE_KERNEL) $@ | $(MKHASH) md5 | awk '{print $$1}' | \
+		tr -d '\n' | dd seek=$$((0x1e0)) of=$@.hdrfactory bs=1 \
+		conv=notrunc 2>/dev/null
+	$(call Build/sercomm-tag-hdr-kernel,$(IMAGE_KERNEL) $(SERCOMM_KERNEL_OFFSET)
+		$(SERCOMM_ROOTFS_OFFSET) $@.hdrkrn1)
+	$(call Build/sercomm-tag-hdr-kernel,$(IMAGE_KERNEL) $(SERCOMM_KERNEL2_OFFSET)
+		$(SERCOMM_ROOTFS2_OFFSET) $@.hdrkrn2)
+	# Hack CRC for Kernel2. In case of flashing to the Sercomm1 slot,
+	# the uboot drain will switch to the Sercomm0 slot. So far, there is
+	# no support for working from both slots.
+	dd if=/dev/zero of=$@.hdrkrn2 bs=1 seek=$$((0x18)) count=4 conv=notrunc \
+		2>/dev/null
+	cat $@.hdrfactory $@.hdrkrn1 $@.hdrkrn2 $(IMAGE_KERNEL) $@ > $@.new
+	mv $@.new $@
+	rm -f $@.hdrfactory $@.hdrkrn1 $@.hdrkrn2
+endef
```

Comment:
Please explain the image format in the commit description.

Reply:
Done.